nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongoRecord.java

```diff
@@ -79,35 +77,29 @@ public class GetMongoRecord extends AbstractMongoQueryProcessor {
         .required(true)
         .build();
 
-    private static final List<PropertyDescriptor> DESCRIPTORS;
-    private static final Set<Relationship> RELATIONSHIPS;
-
-    static {
-        List<PropertyDescriptor> _temp = new ArrayList<>();
-        _temp.add(CLIENT_SERVICE);
-        _temp.add(WRITER_FACTORY);
-        _temp.add(DATABASE_NAME);
-        _temp.add(COLLECTION_NAME);
-        _temp.add(SCHEMA_NAME);
-        _temp.add(QUERY_ATTRIBUTE);
-        _temp.add(QUERY);
-        _temp.add(PROJECTION);
-        _temp.add(SORT);
-        _temp.add(LIMIT);
-        _temp.add(BATCH_SIZE);
-
-        DESCRIPTORS = Collections.unmodifiableList(_temp);
-
-        Set<Relationship> _rels = new HashSet<>();
-        _rels.add(REL_SUCCESS);
-        _rels.add(REL_FAILURE);
-        _rels.add(REL_ORIGINAL);
-        RELATIONSHIPS = Collections.unmodifiableSet(_rels);
-    }
+    private static final List<PropertyDescriptor> PROPERTIES = Stream.of(
```

Comment:
Hi @dan-s1 I'm curious why you sometimes chose to use Stream.of() instead of List.of()?  I understand using Stream when an argument is a collection itself, but this instance and some others appear to be better suited to List.of().

Reply:
@mosermw Excellent question. The `List.of() `method is overloaded and it seems the max number of arguments it can take is 10. Hence where there are more than 10 properties I has to use `Stream.of`.

Reply:
Yeah i'd say unless we're able to move from current practice to one clean model that offers the same semantics with cleaner code then it isn't worth changing.